### PR TITLE
[WIP] Add Xcode8 markup

### PR DESCRIPTION
**Pull request by eimantas**
Also updated the project file for Xcode 8.

### CODE REVIEW

**Comment by tomj** [comment]
@eimantas nice!  I'm on some other stuff today but will try and take a look at this tonight 😄

File: Cartfile.resolved [hunk 1]
```diff
@@ -1,4 +1,4 @@
 github "Quick/Nimble" "v4.0.1"
 github "Quick/Quick" "v0.9.2"
-github "antitypical/Result" "2.0.0"
```
Comment (tomj) [inline Cartfile.resolved]: @eimantas I think this is already in `master` on the mainline from a commit the other day, if so I will clean it up when we PR into the main repo

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0710"
+   LastUpgradeVersion = "0800"
```
Comment (tomj) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: Do you reckon we hold off from this change?   The PR will be going in to `master` which won't be Xcode 8 for a while

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-iOS.xcscheme [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0710"
+   LastUpgradeVersion = "0800"
```
Comment (tomj) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-iOS.xcscheme]: ditto

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-tvOS.xcscheme [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0710"
+   LastUpgradeVersion = "0800"
```
Comment (tomj) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-tvOS.xcscheme]: ditto

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-watchOS.xcscheme [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0710"
+   LastUpgradeVersion = "0800"
```
Comment (tomj) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-watchOS.xcscheme]: ditto

File: ReactiveCocoa/Swift/Action.swift [hunk 6]
```diff
@@ -57,8 +57,14 @@ public final class Action<Input, Output, Error: ErrorType> {
 		return userEnabled && !executing
 	}
 
-	/// Initializes an action that will be conditionally enabled, and create a
```
Comment (tomj) [inline ReactiveCocoa/Swift/Action.swift]: nice catch

File: ReactiveCocoa/Swift/Action.swift [hunk 7]
```diff
@@ -73,8 +79,11 @@ public final class Action<Input, Output, Error: ErrorType> {
 			.map(Action.shouldBeEnabled)
 	}
 
-	/// Initializes an action that will be enabled by default, and create a
+	/// Initializes an action that will be enabled by default, and creates a
 	/// SignalProducer for each input.
+	///
+	/// - parameter execute: A closure that returns the signal producer returned
```
Comment (tomj) [inline ReactiveCocoa/Swift/Action.swift]: Do you think it would be worth doing `- parameters:` even if there is only 1?  I'm not 100% either way, but it might make it a little tidier?

**Comment by tomj** [comment]
@eimantas this all looks awesome 😄 and you've cranked out so much!  Before we push forward much more what do you think about coming up with an example "schema" for comments which we can use?  We could show that to the folks over at ReactiveCocoa and @mdiep and see what they think, and then apply that to all the project

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme (hunk 2 above)
Comment (eimantas) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: Not sure. But @mdiep told that it can be done towards master. So they probably aren't going to do a release for another few months. It's best we coordinate with him just in case.

File: ReactiveCocoa/Swift/Action.swift (hunk 7 above)
Comment (eimantas) [inline ReactiveCocoa/Swift/Action.swift]: IMHO it would make sense if function was bound to have more parameters in the future. In the end though the markup for `parameter` and `parameters` is displayed identically. So I'm good with either .)

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme (hunk 2 above)
Comment (tomj) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: Yeah I'm guessing they want `master` to be able to build with the current Xcode, which is 7.x so we should avoid any references to Xcode 8 in this PR

**Comment by eimantas** [comment]
Well I was experimenting a bit with the markup layout in the code and there are a few things to note:

- It's best to keep lines at 80 characters in length even if we're not on teletype systems anymore (Xcode's marker helps);
- The values for every key (i.e. after the semicolon) should be treated as separate sentences/paragraphs
- If line goes beyond 80 chars in length the next line should be indent at the same level as semicolon on previous line +1 space
- I'm quite ambivalent on whether we should add empty comment line between the markup values:

```
/// parameter foo: Foo for fonction
///
/// returns: bar
```

or 

```
/// parameter foo: Foo for fonction
/// returns: bar
```

On one hand - whitespace is nice, but on the other - I found that documenting functions in `Scheduler.swift` may take up too much space. I might be wrong though.

It's also very hard to distinguish between `note` and `warning`. I used my common-sense judgment to add these to the current documentation.

In any case we should let others chime in on this.

File: ReactiveCocoa/Swift/Action.swift (hunk 7 above)
Comment (tomj) [inline ReactiveCocoa/Swift/Action.swift]: so it would be like:

```
/// - parameter input: A value that will be passed to closure creating the
///                    signal producer.
/// - note: If the action is disabled when the returned SignalProducer is
///         started, the produced signal will send `ActionError.NotEnabled`,
///         and nothing will be sent upon `values` or `errors` for that
///         particular signal.
```
VS
```
/// - parameters:
///	  - input: A value that will be passed to closure creating the
///            signal producer.
/// - note: If the action is disabled when the returned SignalProducer is
///         started, the produced signal will send `ActionError.NotEnabled`,
///         and nothing will be sent upon `values` or `errors` for that
///         particular signal.
```
I like the idea of being able to add more parameters to a function easily so would prefer `- parameters:` if it's ok with you.

Also I'm guessing this was what @mdiep meant when he [said the following](https://github.com/ReactiveCocoa/ReactiveCocoa/issues/2982)

> But I think we should use - parameters: instead of - parameter.

Comment (eimantas) [inline ReactiveCocoa/Swift/Action.swift]: Fair enough, let's use the `parameters` everywhere then.

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme (hunk 2 above)
Comment (eimantas) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: Okay, let's revert this change then.

**Comment by tomj** [comment]
Definitely agree re: the note/warning part - I reckon we can put this out for contributor comment in the main repo.  Let's stick with the white space for now?  I know it takes up a bunch of space but at least for me it's actually an aspect of the RAC documentation which I appreciate (ie: it's well spaced and easy to read).

**Comment by eimantas** [comment]
Alrighty! I'll add more whitespace to the documentation.

**Comment by eimantas** [comment]
And also will convert `parameter` to `parameters`.

**Comment by eimantas** [comment]
Quick question - should there be white space between parameter descriptions?

```
/// - parameters:
///    - foo: Function's foo
///    - bar: Function's bar
```

vs 

```
/// - parameters:
///
///    - foo: Function's foo
///
///    - bar: Function's bar
```

**Comment by tomj** [comment]
@eimantas I reckon the parameters look better with no whitespace - a little easier to read
